lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -384,7 +384,9 @@ def method_missing(symbol, *args)
             end.join("\n")
 
           mixed_in_module = dynamic_extends.find do |mod|
-            mod != constant && public_module?(mod)
+            mod != constant && qualified_name_of(mod) == "::ActiveSupport::Concern" ||
+            (singleton_class_of(constant).ancestors.include?(ActiveSupport::Concern) &&
+             mod == constant.const_get(:ClassMethods))
```

Comment:
We should not be trying to find a module in the extends that matches the criteria we are looking for. That is not the aim of this exercise.

We have two cases:

1. If the constant extends from a class **named** `"ActiveSupport::Concern"` and if the subconstantant `:ClassMethods` is a `Module`, then we should make `mixed_in_module` to be that module. Nothing more to do here and this is what we used to do before.
2. If the above case is not true, then this is a custom implementation of mixing in class methods, then the code above should try to do its best guess finding as before. Nothing more to do.

Reply:
I didn't know we need to cover the else part, thanks Ufuk for explaining, here is what I understood from above: 
```
          extend  = singleton_class_of(constant).ancestors
          if extend.include?(ActiveSupport::Concern)
            mixed_in_module = dynamic_extends.find do |mod|
                mod == constant.const_get(:ClassMethods)
            end
          else
            mixed_in_module = dynamic_extends.find do |mod|
              mod != constant && public_module?(mod)
            end
          end
```
is this the correct translation of what you explained above? 
thnx!